### PR TITLE
Add Color Definitions to ColorSchema Enum (HeatMap)

### PR DESCRIPTION
**Pull request by Nirus2000**
1. **GREEN_YELLOW_RED**: Green for gains, Red for losses, Yellow for neutral values.
2. **GREEN_WHITE_RED**: Green for gains, White for neutral, Red for losses.
3. **GREEN_RED_GREY**: Green for gains, Red for losses, Grey for neutral.
4. **BLUE_ORANGE_GREY**: Blue for stable values, Orange for volatile values, Grey for neutral.
5. **BLACK_YELLOW_WHITE**: Black for extreme gains, Yellow for moderate returns, White for neutral.

The colors are obtained from colors.java

### CODE REVIEW

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java [hunk 1]
```diff
@@ -24,46 +26,92 @@ private ColorSchema(String label)
...
+
+            case BLACK_YELLOW_WHITE -> performance -> {
+                // Performance interpolates between white (neutral), yellow
+                // (moderate), or black (extreme)
+                double p = normalizePerformance(performance);
+
+                RGB color = performance > 0.05
+                                ? Colors.interpolate(Colors.YELLOW.getRGB(), Colors.BLACK.getRGB(),
+                                                (float) ((p - 0.05) / 0.95))
+                                : Colors.interpolate(Colors.WHITE.getRGB(), Colors.YELLOW.getRGB(), (float) p);
```
Comment (buchen) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java:87]: I have to admit, I do not understand the math. The 0.05 are deducted from the normalized 0 -> 1 range. As we normalize between -0.07 and +0.07, there are 0.14 on the range and 5% of that (the deduction) is 0.07.

Plus values that are above 0.07 will be black anyway (interpolation factor 1). As only performance > 0.05 is going into the black branch, the lower bound is 0.8571428571. The ```(p - 0.05) / 0.95)``` will rescale this to 0.8496240602. What difference does it make?


BTW, can you share where you picked up the black-yellow color palette? I looked for materials in the web, but was [not](https://www.datacamp.com/tutorial/seaborn-heatmaps) [really](https://vwo.com/blog/heatmap-colors/) [successful](https://www.atlassian.com/data/charts/how-to-choose-colors-data-visualization) in finding examples of black-yellow.

**Review by buchen — CHANGES_REQUESTED** [review]
Hey @Nirus2000, thanks for enhancing the color palettes.

I am aware that green-red is not good enough because of color blindness and other reasons. Unfortunately, I also do not understand color theory good enough...

Besides the rather smaller comments in the code, I would help me if you can share the sites where you picked up the logic for the color palettes. I tried to search for schemas that are new to me (black - yellow), but could not really find material.

Andreas.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java [hunk 2]
```diff
@@ -24,46 +26,92 @@ private ColorSchema(String label)
...
+                                ? Colors.interpolate(Colors.WHITE.getRGB(), Colors.HEATMAP_DARK_GREEN.getRGB(), (float) p)
+                                : Colors.interpolate(Colors.WHITE.getRGB(), Colors.RED.getRGB(), (float) p);
+
+                return resourceManager.createColor(color);
+            };
+
+            case GREEN_RED_GREY -> performance -> {
+                // Performance interpolates between grey (neutral), green
+                // (positive), or red (negative)
+                double p = normalizePerformance(performance);
```
Comment (buchen) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java:59]: Thinking about it some more, does it really make sense to normalize the performance here?

-0.07 will be normalized to 0, and +0.07 will be normalized to 1.
But then we interpolate depending on the original performance.
So dark gray/green will only be interpolated between 0 and 0.5.
And the gray/red will only be interpolated between 0.5 and 1.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java [hunk 3]
```diff
@@ -24,46 +26,92 @@ private ColorSchema(String label)
...
+            };
+
+            case GREEN_WHITE_RED -> performance -> {
+                // Performance is normalized and interpolated between white
+                // (neutral) and green (positive) or red (negative)
+                double p = Math.min(MAX_PERFORMANCE, Math.abs(performance)) / MAX_PERFORMANCE;
+
+                RGB color = performance > 0f
+                                ? Colors.interpolate(Colors.WHITE.getRGB(), Colors.HEATMAP_DARK_GREEN.getRGB(), (float) p)
+                                : Colors.interpolate(Colors.WHITE.getRGB(), Colors.RED.getRGB(), (float) p);
```
[the remark this answers is not in the document]
Comment (Nirus2000) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java]: Done and updated the interpolate method to clamp the factor parameter between 0 and 1.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java [hunk 4]
```diff
@@ -24,46 +26,92 @@ private ColorSchema(String label)
...
+
+                return resourceManager.createColor(color);
+            };
+
+            case GREEN_RED_GREY -> performance -> {
```
[the remark this answers is not in the document]
Comment (Nirus2000) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java]: done

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java [hunk 5]
```diff
@@ -24,46 +26,92 @@ private ColorSchema(String label)
...
+                // Performance interpolates between grey (neutral), green
+                // (positive), or red (negative)
+                double p = normalizePerformance(performance);
+
+                RGB color = performance > 0 ? Colors.interpolate(Colors.GRAY.getRGB(), Colors.HEATMAP_DARK_GREEN.getRGB(), (float) p)
+                                : Colors.interpolate(Colors.GRAY.getRGB(), Colors.RED.getRGB(), (float) p);
```
[the remark this answers is not in the document]
Comment (Nirus2000) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java]: We can add the following change
```
case GREEN_YELLOW_RED -> performance -> {
    double p = normalizePerformance(performance);
    float hue = (float) (p * 120); // 0 (red) bis 120 (green)

    // Increase intensity for the lower values
    if (p < 0.5) {
        hue *= 2;
    }
    
    return resourceManager.createColor(new RGB(hue, 0.9f, 1f));
};

**Comment by Nirus2000** [comment]
Hello @buchen 
I have implemented all your changes as best I could.

I have the following comments.
If we start with the `Colors.theme().defaultBackground().getRGB()`then the colors are interpolated in the DarkTheme, which means that the translations of the HeatMap color schemes are of course no longer correct. 
How do we want to solve this problem ❓ 🤔 

❓ It also remains to be seen whether we can define the heat maps more clearly than just using the color scheme. (see description below)
 
---

### Green-Yellow-Red:
This combination is often used to represent positive, neutral and negative performance.

### Green-White-Red:
This color scale is good for showing returns moving into a neutral zone.
When analyzing individual stocks, this heat map can help show investors how a stock has moved from neutral (white) to positive (green) and negative (red) returns. It is particularly useful if you want to differentiate between losses and gains.
[TradingView - Performance Heatmap](https://www.tradingview.com/script/bBPyKI9b-Stock-Market-Indices-Correlations-Heatmap/)

### Green-Gray-Red:
This chart is suitable for visualizing performance in the context of risk.
In a risk management dashboard, green boxes could represent stocks that combine high risk with positive performance, while red boxes represent high risk with poor performance. Gray fields could indicate neutral investments.
[finviz](https://finviz.com/map.ashx?st=w1)

### Blue-Gray-Orange:
This color scale is particularly useful for representing stable versus volatile assets.
In a currency market heat map, blue could represent stable currencies, while orange would represent volatile currencies. Gray boxes could represent a stable background, meaning that no significant movements have taken place.
[A1 - Trading](https://www.a1trading.com/currency-heatmap/)

### Yellow-White-Black:
This combination is useful to show moderate and extreme changes.
In a commodity price heat map, yellow could represent moderate price changes, while black shows extreme price fluctuations. This visualization helps traders prepare for potential market movements.

**Pull request closed without merging**